### DISCUSSION TITLE
proxy to devServer.proxy.target

### DISCUSSION THREAD

**Discussion by AllForNothing** [chimurai/http-proxy-middleware#642] [Q&A] [answered]
### Describe the feature you'd love to see
```
module.exports = {
  //...
   devServer: {
      proxy: {
        '/api': {
         target: 'https://other-server.example.com',
         secure: false,
      },
    },
  },
};
```

for the above config,   my computer can not visit the target address directly,  so I have to set a proxy for the target address.

In the Terminal,   I can  get the response by  "curl"  the target address  with the below setting
export https_proxy=http://10.xxx.xx.xx:3128
export http_proxy=http://10.xxx.xx.xx:3128
export no_proxy="127.0.0.1,localhost,10.*.*.*"

But when   I  run "yarn start“,  the API requests to the target address directly will fail because the above proxy setting does not work for the devServer.

So I'm wondering if there is a way to set a proxy to the target address like this:
```
module.exports = {
  //...
  devServer: {
    proxy: {
         '/api': {
          target: 'https://other-server.example.com',
          httpsProxyForTarget: "http://10.xxx.xx.xx:3128"
          httpProxyForTarget: "http://10.xxx.xx.xx:3128"
         secure: false,
      },
    },
  },
};
```

Thanks!!!!

**Comment by chimurai** [comment, marked as answer]
Think you're looking for this recipe:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/corporate-proxy.md#corporate-proxy-support

**Reply by AllForNothing** [reply]
@chimurai Thanks a lot. It surely solved my problem
